src/HttpExtractor.php

```diff
@@ -21,7 +25,23 @@ public function __construct(
 
     public function extract(UriInterface $httpSource, string $filesystemDestination): void
     {
-        $this->client->get($httpSource, ['sink' => $filesystemDestination]);
+        try {
+            $this->client->get($httpSource, ['sink' => $filesystemDestination]);
+        } catch (ClientException|ServerException $e) {
+            throw new UserException(sprintf(
+                'Server returned HTTP %s for "%s"',
+                $e->getCode(),
+                (string)$httpSource
+            ), 0, $e);
+        } catch (ConnectException $e) {
+            if ($e->getHandlerContext()['errno'] !== CURLE_COULDNT_RESOLVE_HOST) {
+                throw $e;
+            }
+            throw new UserException(sprintf(
+                'Unknown hostname in "%s"',
+                (string)$httpSource
+            ), 0, $e);
```

Comment:
Čistě teoretickej dotaz. Je volání 

```
new \Exception($message, 0, $e);
```
něčím odlišný od
```
new \Exception($message, null, $e);
```

Osobně použivám tu druhou variantu, ale to jsem odněkud převzal a neřešil jsem to.

Reply:
int $code, http://php.net/manual/en/exception.construct.php

Reply:
aha, `declare(strict_types=1);`!

Reply:
A taky je to výchozí hodnota.